src/main/java/org/testng/internal/TestMethodWorker.java

```diff
@@ -53,13 +55,22 @@ public TestMethodWorker(IInvoker invoker,
                           List<IClassListener> listeners)
   {
     m_invoker = invoker;
-    m_methodInstances = testMethods;
     m_suite = suite;
     m_parameters = parameters;
     m_groupMethods = groupMethods;
     m_classMethodMap = classMethodMap;
     m_testContext = testContext;
     m_listeners = listeners;
+    List<IMethodInstance> localMethodInstances = Lists.newArrayList(testMethods);
```

Comment:
The order is supposed to be managed by the graph when we add edges.

At the opposite, I'm trying to remove all other locations where an ordering is done: https://github.com/cbeust/testng/pull/1352